app/controllers/pages_controller.rb

```diff
@@ -1,6 +1,12 @@
 class PagesController < ApplicationController
-
   def current_styleguide
   end
 
+  def show
+    page = current_website.pages.published.find_by!(slug: params[:page])
+    @body = page.published_body
+    render layout: "themes/#{current_website.theme}"
+  rescue ActiveRecord::RecordNotFound
+    redirect_to '/404'
```

Comment:
This is using the cfp_app 404 page but we will ultimately want to create a version that works better for the front end.